apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -640,8 +658,13 @@ export default class InstallManager {
    * to the command-line.
    */
   public pushConfigurationArgs(args: string[]): void {
-    args.push('--cache', this._rushConfiguration.npmCacheFolder);
-    args.push('--tmp', this._rushConfiguration.npmTmpFolder);
+    if (this._rushConfiguration.packageManager === 'npm') {
+      args.push('--cache', this._rushConfiguration.npmCacheFolder);
+      args.push('--tmp', this._rushConfiguration.npmTmpFolder);
+    } else if (this._rushConfiguration.packageManager === 'pnpm') {
+      args.push('--store', this._rushConfiguration.pnpmStoreFolder);
+      args.push('--no-lock');
```

Comment:
>'--no-lock' [](start = 16, length = 11)

Why do we need this?  It prints a big warning.

Explain in a code code comment

Reply:
Would it be better to delete PNPM's lock file before starting?

---
In reply to: [153378631](https://github.com/Microsoft/web-build-tools/pull/449#discussion_r153378631) [](ancestors = 153378631)

Reply:
Ideally we would look and see if the lock file exists, if it does, we should delete the entire store. If the lock file does not exist, we are probably in a good state to continue. Likewise, if someone wants to do a clean install, we could technically not delete the store if the lockfile doesn't exist (although this would likely break the semantics of a --clean flag)

---
In reply to: [153378665](https://github.com/Microsoft/web-build-tools/pull/449#discussion_r153378665) [](ancestors = 153378665,153378631)